grails-app/services/org/broadinstitute/orsp/StorageProviderService.groovy

```diff
@@ -163,21 +163,22 @@ class StorageProviderService implements Status {
      * @param files The multipart files
      * @return
      */
-    StorageDocument saveMultipartFile(String displayName, String userName, String issueKey, String type, MultipartFile file) {
+    StorageDocument saveMultipartFile(String displayName, String userName, String issueKey, String type, MultipartFile file, Long consentCollectionLinkId) {
         StorageDocument document = new StorageDocument(
-                projectKey: issueKey,
+                projectKey: consentCollectionLinkId == null ? issueKey : null,
                 fileName: file.originalFilename,
                 fileType: type,
                 mimeType: file.contentType,
                 uuid: UUID.randomUUID().toString(),
                 creator: displayName,
                 username: userName,
                 creationDate: new Date(),
-                status: DocumentStatus.PENDING.status
+                status: DocumentStatus.PENDING.status,
+                consentCollectionLinkId: consentCollectionLinkId
         )
         if (saveStorageDocument(document, file.getInputStream())) {
             persistenceService.saveEvent(
-                    document.projectKey,
+                    issueKey,
```

Comment:
It could be the case that this is null now that project and ccl_id are each nullable values.